### ISSUE TITLE
Split transaction with multiple revenue/expense accounts #613

### ISSUE THREAD

**Issue by lungj** [firefly-iii/firefly-iii#2604]
**Description**
1. I would like Firefly III to allow one debit/credit from an expense account to have multiple destinations. This is not a duplicate of #613.
2. I would like Firefly III to allow withdrawals from revenue accounts directly to expense accounts.

**Solution**
Presently, when I receive my paycheque, several taxes and miscellanea are deducted automatically and remitted to the government. When filing taxes, I would like to be able to track how much has been paid for each of those deductions so that I can reclaim them on my taxes.

For example, my monthly paycheque might give me a gross income of $1000 of which $200 is remitted to the government in taxes and I receive the other $800 direct deposited into my account (so, annually, $12000 gross income, $2400 remitted for taxes, and $9600 deposited to my bank). At the end of the year, when calculating my taxes, it might work out that I owe $2000 in taxes which means my company has over-remitted taxes to the government and I am owed $400 back. I would like to be able to track this easily. I've had paycheques that have had up to, I believe, 6 such withholdings/remittances before reaching my bank.

**What are alternatives?**
The way that I would do this without any extra features is one of two ways:

 * Since I cannot directly send money from a revenue account to an expense account, I would create an asset account that does not contribute to my net worth (it's a little weird that I'm counting government money in my asset account). At the end of the year, I can add a balancing withdrawal from that asset account to a government expense account which, when I receive my tax refund, comes back from the same expense account.
 * Repeat the above step once for each asset account for each of the types of withholdings are on my paycheque and repeat several times OR create one general paycheque deductions account, add a second transaction for each paycheque for the sum of the deductions amount, and then create a third transaction for each paycheque with multiple payees from the common deductions account.

**Additional context**

**Bonus points**
Thank you kindly!

**Comment by JC5** [comment]
No. The goal of Firefly is to track your personal finances. Money you don't have won't be a part of the tool. You'll have to use your alternatives or use another tool.

**Comment by lungj** [comment]
Are your personal taxes not part of your personal finances?

**Comment by JC5** [comment]
Not the way you're describing it. As far as I can see it's your company withholding the taxes.

**Comment by lungj** [comment]
Also, part of my paycheque automatically goes towards my pension (it's automatic in Canada).

**Comment by lungj** [comment]
Yes, it's my company withholding taxes. Since you're the one doing the work, you're the one calling the shots :) That being said, tax planning is listed as one of the 10 focuses of personal finance on Wikipedia and I'm explicitly trying to figure out how to manage personal cash flow come tax day! Anyway, thanks for responding. I think I will have to take your suggestion and go back to my status quo of GnuCash.

**Comment by lungj** [comment]
I stuck with it :) For anyone else running into this, here's what I did:

* I ended up creating multiple asset accounts that contribute to net worth and used recurring transactions to make things easier (to avoid doing this for each paycheque received; I will have to change this since the withholdings change periodically throughout the year).
* Money from my salary is put into the different government-mandated contribution categories by the recurring transactions and then a balancing transaction is used to remit to the government (i.e., the transactions end up passing through Firefly, even though I don't have direct control over the account until tax season), also via recurring transaction.
* When tax season comes around, I'll be able to see exactly how much has gone through each account (by year) for filing purposes. I can then budget my cash flow for the refund for the remainder of the year, even though I won't know exactly when I'll get it back.
* When the refund is generated, I'll add a government deposit into that account + withdrawal to my bank account from the virtual asset.
